Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IConceptEntryRepository.java

```diff
@@ -0,0 +1,42 @@
+package edu.asu.conceptpower.app.repository;
+
+import org.springframework.stereotype.Repository;
+
+import edu.asu.conceptpower.app.model.ConceptEntry;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
+
+/**
+ * Concept Entry repository
+ *  
+ * @author Keerthivasan Krishnamurthy
+ * 
+ */
+@Repository
+public interface IConceptEntryRepository extends PagingAndSortingRepository<ConceptEntry, String>, JpaSpecificationExecutor{
+    
+    List<ConceptEntry> findAll();
```

Comment:
I don't think this method is needed, it should be there by default.

Reply:
Yeah, it's there by default. But, the return value is Iterable<>. I've added that as a separate method to return as a List<>.

Reply:
Well, it's probably returning an iterable, so not all data will be added into memory at once. Why don't you use the iterable?

Reply:
Sure we can do that. I changed it to return in List<> because we can avoid changes to the underlying Manager classes consuming this service.

Reply:
where is that actually needed?

Reply:
Umm for instance taking ConceptList, it will be used in Concept wrapper service, Concept Add/Delete controllers and it's dependant class Models.

Reply:
why do they need to find all concepts? aren't all lists paginated?

Reply:
No, the lists aren't paginated

For add/edit concepts, I think we are using it to populate the dropdown